### PR TITLE
ECDC-3756: Update produced bytes metric

### PR DESCRIPTION
**Pull request by amues**
### Issue reference / description

ECDC-3756: add updated Kafka and produce metrics to all modules.

## Checklist for submitter

- [ ] Check for conflict with integration test
- [ ] Unit tests pass

---

### Nominate for Group Code Review

- [ ] Nominate for code review

### CODE REVIEW

File: src/modules/trex/TREXBase.cpp [hunk 1]
```diff
@@ -104,14 +104,28 @@ TrexBase::TrexBase(BaseSettings const &settings) : Detector(settings) {
...
+  Stats.create("kafka.config_errors", Counters.KafkaStats.config_errors);
+  Stats.create("kafka.produce_bytes_ok", Counters.KafkaStats.produce_bytes_ok);
+  Stats.create("kafka.produce_bytes_error", Counters.KafkaStats.produce_bytes_error);
+  Stats.create("kafka.produce_calls", Counters.KafkaStats.produce_calls);
+  Stats.create("kafka.produce_no_errors", Counters.KafkaStats.produce_no_errors);
   Stats.create("kafka.produce_errors", Counters.KafkaStats.produce_errors);
+  Stats.create("kafka.err_unknown_topic", Counters.KafkaStats.err_unknown_topic);
+  Stats.create("kafka.err_queue_full", Counters.KafkaStats.err_queue_full);
+  Stats.create("kafka.err_other", Counters.KafkaStats.err_other);
   Stats.create("kafka.ev_errors", Counters.KafkaStats.ev_errors);
```
Comment (tbukovics) [inline src/modules/trex/TREXBase.cpp:122]: Remove these old Kafka stats since you introduced the new ones.

**Review by tbukovics — CHANGES_REQUESTED** [review]
Some old counters you forgot in place and I also asked for clarification. 
- Also as we dicussed plase add a check into the integration test json, where you compare that NOT OK transmitted bytes are equal to zero and test that OK transmitted bytes are bigger then 0.

**Review by tbukovics — CHANGES_REQUESTED** [review]
I made some request for modification and also asked for some clarification.

Also do not forget to add to the integration tests to check on trasmitted byte counters:
- NOT OK is == 0
- OK is > 0

File: src/common/kafka/EV44Serializer.cpp [hunk 2]
```diff
@@ -101,7 +101,7 @@ size_t EV44Serializer::produce() {
       ProduceFunctor(Buffer_,
                      Event44Message_->reference_time()->data()[0] / 1000000);
     }
-    // \todo should this be inside the previous for?
+    // \todo change to new producer metrics and remove
```
[the remark this answers is not in the document]
Comment (amues) [reply src/common/kafka/EV44Serializer.cpp:113]: The `TxBytes` counter (and associated metric `transmit.bytes`) duplicate functionality offered by the new producer metrics we have added.

I wanted to merge this pull request first before I address this, as this branch has existed for some time and the size of the pull request was growing, and removing the `transmit.bytes` might cause some compatibility issues with tests and dashboards; I'm considering doing this in a new branch after merging this, but let me know if you think we should do everything here.

File: src/common/kafka/Producer.h [hunk 3]
```diff
@@ -68,6 +68,8 @@ class Producer : public ProducerBase, public RdKafka::EventCb {
     int64_t ev_others;
```
[the remark this answers is not in the document]
Comment (amues) [reply src/common/kafka/Producer.h:68]: These counters are complementary to the newly added ones; we don't use them all the time, but they can provide details on errors, so I would suggest keeping them.

File: src/modules/timepix3/Timepix3Base.cpp [hunk 4]
```diff
@@ -77,16 +77,28 @@ Timepix3Base::Timepix3Base(BaseSettings const &settings)
...
+  Stats.create("kafka.produce_bytes_error", Counters.KafkaStats.produce_bytes_error);
   Stats.create("kafka.produce_calls", Counters.KafkaStats.produce_calls);
-  Stats.create("kafka.produce_errors", Counters.KafkaStats.produce_errors);
-  Stats.create("kafka.produce_config_errors", Counters.KafkaStats.config_errors);
   Stats.create("kafka.produce_no_errors", Counters.KafkaStats.produce_no_errors);
+  Stats.create("kafka.produce_errors", Counters.KafkaStats.produce_errors);
+  Stats.create("kafka.err_unknown_topic", Counters.KafkaStats.err_unknown_topic);
+  Stats.create("kafka.err_queue_full", Counters.KafkaStats.err_queue_full);
+  Stats.create("kafka.err_other", Counters.KafkaStats.err_other);
   Stats.create("kafka.ev_errors", Counters.KafkaStats.ev_errors);
```
[the remark this answers is not in the document]
Comment (amues) [reply src/modules/timepix3/Timepix3Base.cpp:95]: These counters are complementary to the new ones (see comment for Producer.h).

File: src/modules/trex/TREXBase.cpp (hunk 1 above)
Comment (amues) [reply src/modules/trex/TREXBase.cpp:122]: These counters are complementary to the new ones (see comment for Producer.h).

File: src/common/kafka/EV44Serializer.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (tbukovics) [reply src/common/kafka/EV44Serializer.cpp:113]: OK then. We can leave this.

File: src/common/kafka/Producer.h (hunk 3 above)
[the remark this answers is not in the document]
Comment (tbukovics) [reply src/common/kafka/Producer.h:68]: ok

**Review by tbukovics — APPROVED** [review]
